packages/flutter_test/lib/src/binding.dart

```diff
@@ -1044,14 +1044,32 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
   @override
   void _verifyInvariants() {
     super._verifyInvariants();
-    assert(
-      _currentFakeAsync.periodicTimerCount == 0,
-      'A periodic Timer is still running even after the widget tree was disposed.'
-    );
-    assert(
-      _currentFakeAsync.nonPeriodicTimerCount == 0,
-      'A Timer is still pending even after the widget tree was disposed.'
-    );
+    try {
```

Comment:
I think the idiom more commonly used here would be something like:

```dart
assert(() {
  if (_currentFakeAsync.periodTimerCount != 0) {
    // Print error/message(s)
    return false;
  }
  // ...
  return true;
}());
```

Reply:
I was trying to preserve the existing `AssertionError`s and their messages.

I suppose now that the debugging information should report whether the pending timers are periodic or non-periodic (albeit in a less friendly form), that distinction is less important.